cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -548,6 +555,34 @@ func (s *State) RemoveIntercept(interceptID string) bool {
 	return didDelete
 }
 
+// GetInterceptAPIKeys returns a slice of all the apiKeys associated with intercept IDs.
+// Since we use this fuction as a last resort if we need to garbage collect intercepts
+// when there are no active sessions, we just return the slice since we aren't picky about
+// which apiKey is used.
+func (s *State) GetInterceptAPIKeys() []string {
+	s.mu.Lock()
+	apiKeys := make([]string, 0, len(s.interceptAPIKeys))
+	for _, key := range s.interceptAPIKeys {
+		apiKeys = append(apiKeys, key)
+	}
+	s.mu.Unlock()
+	return apiKeys
+}
```

Comment:
Probably easier to just return the first non-empty API key?  Simplify the code here, simplify the code in systema.go, avoid a few allocations.